api/channel.go

```diff
@@ -397,10 +397,15 @@ func updateChannelPurpose(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getChannels(c *Context, w http.ResponseWriter, r *http.Request) {
+	// check if the team exist
+	if result := <-Srv.Store.Team().Get(c.TeamId); result.Err != nil {
```

Comment:
0/5 on where, but not sure you can add it to context.go since it has unauthenticated calls too.

I think it's fine put it at the top of the function, my main concern was you should be able to avoid the DB hit for `Srv.Store.Team().Get(...)` since which teams you can access is cached in memory via your session.